safehttp/form.go

```diff
@@ -0,0 +1,229 @@
+// the getter functions.
+type Form struct {
+	values map[string][]string
+	err    error
+}
+
+// Int64 checks whether key paramKey maps to any query or form parameter
+// values. In case it does, it will try to convert the first value to a 64-bit
+// integer and return it. If there are no values associated with paramKey, it
+// will return the default value. If the first value is not an integer, it will
+// return the default value and set the Form error field.
+func (f *Form) Int64(paramKey string, defaultValue int64) int64 {
+	vals, ok := f.values[paramKey]
+	if !ok {
+		return defaultValue
+	}
+	paramVal, err := strconv.ParseInt(vals[0], 10, 64)
+	if err != nil {
+		f.err = err
+		return defaultValue
+	}
+	return paramVal
+}
+
+// Uint64 checks whether key paramKey maps to any query or form parameter
+// values. In case it does, it will try to convert the first value to an
+// 64-bit unsigned integer and return it. If there are no values associated with
+// paramKey, it will return the default value. If the first value is not an
+// unsigned integer, it will return the default value and set the Form
+// error field.
+func (f *Form) Uint64(paramKey string, defaultValue uint64) uint64 {
+	vals, ok := f.values[paramKey]
+	if !ok {
+		return defaultValue
+	}
+	paramVal, err := strconv.ParseUint(vals[0], 10, 64)
+	if err != nil {
+		f.err = err
+		return defaultValue
+	}
+	return paramVal
+}
+
+// String checks whether key paramKey maps to any query or form parameter
+// values. In case it does, it will return the first value. If it doesn't, it
+// will return the default value.
+func (f *Form) String(paramKey string, defaultValue string) string {
+	vals, ok := f.values[paramKey]
+	if !ok {
+		return defaultValue
+	}
+	return vals[0]
+}
+
+// Float64 checks whether key paramKey maps to any query or form parameter
+// values. In case it does, it will try to convert the first value to a float
+// and return it. If there are no values associated with paramKey, it will
+// return the default value. If the first value is not a float, it will return
+// the default value and set the Form error field.
+func (f *Form) Float64(paramKey string, defaultValue float64) float64 {
+	vals, ok := f.values[paramKey]
+	if !ok {
+		return defaultValue
+	}
+	paramVal, err := strconv.ParseFloat(vals[0], 64)
+	if err != nil {
+		f.err = err
+		return defaultValue
+	}
+	return paramVal
+}
+
+// Bool checks whether key paramKey maps to any query  or form parameter
+// values. In case it does, it will try to convert the first value to a boolean
+// and return it. If there are no values associated with paramKey, it will
+// return the default value. If the first value is not a boolean, it will return
+// the default value and set the Form error field.
+func (f *Form) Bool(paramKey string, defaultValue bool) bool {
+	vals, ok := f.values[paramKey]
+	if !ok {
+		return defaultValue
+	}
+	switch vals[0] {
+	case "true":
```

Comment:
I've noticed but chose to opt for the obvious cases for now, as we didn't discussed the other ones yet. I'll open an issue since this PR is pretty big already.

Reply:
Please file an issue to reconsider this. Let's keep just `true` and `false` for now.

Reply:
Done :)